Quantum Constraint Solver with dreal
gates list: ['sx(0)']
target probability: [0.08152173913043478, 0.07608695652173914, 0.016304347826086956, 0.043478260869565216, 0.03804347826086957, 0.09782608695652174, 0.05434782608695652, 0.10326086956521739, 0.043478260869565216, 0.10326086956521739, 0.043478260869565216, 0.09239130434782608, 0.09239130434782608, 0.08152173913043478, 0.010869565217391304, 0.021739130434782608]
flag: ==
Exploring.......................
Traceback (most recent call last):
  File "/home/xiashangzhou/quantum_constraint_solver/QCS.py", line 40, in <module>
    result, time, already_result = quantum_constraint_solver(num_qbits=options.qbit_num,
  File "/home/xiashangzhou/quantum_constraint_solver/quantum_solver.py", line 175, in quantum_constraint_solver
    result, time_full = circuit.prove(overapproximation=False, matrix_method= matrix_method)
  File "/home/xiashangzhou/quantum_constraint_solver/symqv/models/circuit.py", line 95, in prove
    return self._prove_state_model(dump_smt_encoding,
  File "/home/xiashangzhou/quantum_constraint_solver/symqv/models/circuit.py", line 348, in _prove_state_model
    result = subprocess.run(command, capture_output=True, timeout=5000)
  File "/home/xiashangzhou/miniconda/envs/quantum_concolic/lib/python3.9/subprocess.py", line 507, in run
    stdout, stderr = process.communicate(input, timeout=timeout)
  File "/home/xiashangzhou/miniconda/envs/quantum_concolic/lib/python3.9/subprocess.py", line 1134, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
  File "/home/xiashangzhou/miniconda/envs/quantum_concolic/lib/python3.9/subprocess.py", line 1996, in _communicate
    self._check_timeout(endtime, orig_timeout, stdout, stderr)
  File "/home/xiashangzhou/miniconda/envs/quantum_concolic/lib/python3.9/subprocess.py", line 1178, in _check_timeout
    raise TimeoutExpired(
subprocess.TimeoutExpired: Command '['/opt/dreal/4.21.06.2/bin/dreal', '--precision', '0.005', '/tmp/tmpa2j5z8gz.smt2']' timed out after 5000 seconds
